prql-compiler/src/utils.rs

```diff
@@ -108,3 +108,29 @@ impl<T> OrMap<T> for Option<T> {
         }
     }
 }
+
+pub trait Pluck<T> {
+    fn pluck<R, F>(&mut self, f: F) -> Vec<R>
+    where
+        F: Fn(T) -> Result<R, T>;
+}
+
+impl<T> Pluck<T> for Vec<T> {
```

Comment:
Am I right that this returns the items in the Vec that return `Ok` when the function is run, and keeps the items that return `Err`? (I can add this as a docstring if so)